cmake/kokkos_tpls.cmake

```diff
@@ -21,7 +21,7 @@ endfunction()
 
 kokkos_tpl_option(HWLOC Off TRIBITS HWLOC)
 kokkos_tpl_option(CUDA ${Kokkos_ENABLE_CUDA} TRIBITS CUDA)
-if(KOKKOS_ENABLE_HIP AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC)
+if(KOKKOS_ENABLE_HIP)
```

Comment:
If you remove the ` KOKKOS_CXX_COMPILER_ID` condition, you can merge this `if ... else` with the one just below

Reply:
done in https://github.com/kokkos/kokkos/pull/8747/commits/21510020a6f489a4162a786cf7ddd26cab4b120b